### DISCUSSION TITLE
Has the official considered using nestjs as the data layer? I believe many people will like it.

### DISCUSSION THREAD

**Discussion by JinYuSha0** [remix-run/remix#8303] [Proposals]
I like remix because it's concise enough, but only for SSR. If it can also be combined with the ecology of nestjs, the experience will be perfect.I developed a library that can connect remixjs as the view layer in nestjs at a very low cost, hope someone likes it.

Supports dependency injection, as well as various decorators and local or global interceptors, filters, pipes and guards

[nest-remix](https://github.com/JinYuSha0/nest-remix)

```typescript
// backend
@Injectable()
@UseGuards(new UserAuthGuard())
export class IndexBackend {
  constructor(private readonly appService: AppService) {}

  @Loader()
  loader(
    @RemixArgs() args: LoaderFunctionArgs,
    @Req() req: Request,
    @Query('name') name?: string,
  ) {
    return { message: this.appService.getHello(name) + ', now: ' + Date.now() };
  }

  @Action()
  action(@Body() body: LoginDto) {
    if (this.appService.login(body)) {
      return redirect('/foo');
    }
    return { msg: 'The username or password is incorrect' };
  }
}

export const useIndexLoader = (args: LoaderFunctionArgs) =>
  useLoader(IndexBackend)(args);

export const useIndexAction = (args: ActionFunctionArgs) =>
  useAction(IndexBackend)(args);
```

```typescript
// frontend
export const loader: LoaderFunction = (...args) => {
  return useIndexLoader(...args);
};

export const action: ActionFunction = (...args) => {
  return useIndexAction(...args);
};

export default function Index() {
  return <div><div>
}
```

I have a complete example,If anyone is interested, you can try running it, It feels like react combined with redux😁

<img width="544" alt="image" src="https://github.com/remix-run/remix/assets/20939091/b0e7059e-2bd0-48eb-b69c-b233ce45534c">

**Comment by JinYuSha0** [comment]
@sergiodxa Is there any chance we can discuss this together?

**Reply by sergiodxa** [reply]
Feel free to discuss here, if you want [a private 1-1 you can through my GitHub Sponsor tiers](https://github.com/sponsors/sergiodxa/sponsorships?tier_id=83905), but if the reason you want to discuss is to get this into Remix I'm not part of the Remix team to do that decission.
